lib/backend/sanitize.go

```diff
@@ -36,11 +37,25 @@ const errorMessage = "special characters are not allowed in resource names, plea
 var allowPattern = regexp.MustCompile(`^[0-9A-Za-z@_:.\-/+]*$`)
 
 // denyPattern matches some unallowed combinations
-var denyPattern = regexp.MustCompile(`//`)
+var denyPatterns = []string{
+	"//",
+	"..",
+}
```

Comment:
I can confirm that `.` is an additional case we need to cover here.  I was unable to reproduce this issue with a combination of `.` and `/`, but if we can be more restrictive that would be best.

Reply:
This definitely isn't exhaustive. `.` is valid, however, because we create `.locks` pretty regularly. The other two will be caught by this because this is just doing a simple `bytes.Contains`.

Reply:
We need to protect against the case where `.` is the entire name.  I edited my prior comment, but in general I think we should just require at least one alpha or numeric character to be present.

Reply:
I'm thinking that we should adjust this so that it's the following:

* `.` is the full name
* `..` is the full name
* `../` is found anywhere in the string
* `//` is found anywhere in the string.

Thoughts?

Reply:
Have we considered just running `path.Clean()` on this thing and seeing if the output matches the input?

Reply:
It sounds like what we are trying to say is:

* A path component can't be empty (`//` and variants)
* A path component can't be solely composed of dots or other special characters (must have a letter?)

I don't mind landing the change, but I'm unclear why we care. It's not like these are actual filepaths or would be resolved as such. Could anyone give a bit more background on this?

Reply:
I tried `path.Clean()`. I imagine I could make it work, but it doesn't quite work as well when there's no leading slash. I'm a little hesitant to munge the string too much to force a match.

Reply:
I've adjusted this to use some fancier regexes that will allow `..string` as a name, which the UI is able to parse without issue.